c2cgeoform/schema.py

```diff
@@ -20,5 +20,22 @@ def __init__(self, *args, **kw):
         self.dbsession = deferred_dbsession
 
     def objectify(self, dict_, context=None):
-        context = super().objectify(dict_, context)
-        return self.dbsession.merge(context)
+        if isinstance(dict_, self.inspector.class_):
+            return dict_
+        return super().objectify(dict_, context)
+
+
+def manytomany_validator(node, cstruct):
+    dbsession = node.bindings['dbsession']
+    class_ = node.children[0].inspector.class_
+    for i, dict_ in enumerate(cstruct):
+        query = dbsession.query(class_)
```

Comment:
Should be easier to read with an inner function call (function which can have a very significant name), but I understand that there is a difficulty with in place replacement.
https://stackoverflow.com/questions/3000461/python-map-in-place

Reply:
I project to use a new specific node class for this case, so I will keep it like that for now, but python-map-in-place in mind.